worker/create_batches_worker.go

```diff
@@ -274,15 +274,24 @@ func (b *CreateBatchesWorker) Process(message *workers.Msg) {
 		b.checkErr(&msg.Job, err)
 	}
 
-	// pull from db, send to control and send to kafta
+	// pull from db, send to control and send to kafka
 	b.processIDs(ids, &msg)
 
 	completedParts := b.setAsComplete(msg.Part, &msg.Job)
 
 	if completedParts == msg.TotalParts {
 		ids = b.getSplitedIds(msg.TotalParts, &msg.Job)
+
 		b.processIDs(ids, &msg)
-		msg.Job.TagSuccess(b.Workers.MarathonDB, nameCreateBatches, "finished")
+
+		if msg.Job.TotalUsers == 0 {
+			_, err := b.Workers.MarathonDB.Model(&msg.Job).Set("completed_at = ?", time.Now().UnixNano()).Where("id = ?", msg.Job.ID).Update()
```

Comment:
In the flow with valid user ids, when is this set?

Reply:
In the `processBatch` method ([ref](https://github.com/topfreegames/marathon/blob/2b403dcff57b3f2f7df1c2c797583bfbf4dbeb57/worker/create_batches_worker.go#L107)), if it finds users, it will call `updateTotalUsers`([ref](https://github.com/topfreegames/marathon/blob/2b403dcff57b3f2f7df1c2c797583bfbf4dbeb57/worker/create_batches_worker.go#L137)), which updates this

Reply:
Giving you created the function that Thiago suggested, WDYT about also calling it in the updateTotalUsers method as well?

Reply:
You mean to set the `completed_at` field every time `updateTotalUsers` is called? If so, that wouldn't work, because the `TotalUsers` field being updated does not mean the job is completed. 

The `updateTotalUsers` method is called multiple times, once for every batch that has at least one known user id, and updates the database by adding up the amount of users being processed in the batch.

Reply:
Makes sense, thanks for the explanation!